### PR TITLE
EN-136: Now pto days calculations takes into account employees country

### PR DESCRIPTION
**Pull request by Paez99**
# Description :pencil:

Added country as a mandatory field for employee creation
Now pto days calculations takes into account employees country
Added unit test
Added country-employee relationship
Set country field in employees table as not null and also added a script that will fix any employee that didnt have a country set and set their country to argentina

## Link to ticket :link:

https://entropyteam.atlassian.net/browse/EN-136

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: src/main/java/com/entropyteam/entropay/employees/services/PtoService.java [hunk 1]
```diff
@@ -6,6 +6,8 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.UUID;
+
+import com.entropyteam.entropay.employees.repositories.*;
```
Comment (FedericoA15) [inline src/main/java/com/entropyteam/entropay/employees/services/PtoService.java:10]: Minor changes in imports repositories

**Review by FedericoA15 — APPROVED** [review]
Good jon

File: src/main/java/com/entropyteam/entropay/employees/models/Employee.java [hunk 2]
```diff
@@ -7,13 +7,7 @@
...
 import com.entropyteam.entropay.employees.dtos.EmployeeDto;
 
-import javax.persistence.Entity;
-import javax.persistence.ManyToMany;
-import javax.persistence.OneToMany;
-import javax.persistence.Table;
-import javax.persistence.FetchType;
-import javax.persistence.JoinTable;
-import javax.persistence.JoinColumn;
+import javax.persistence.*;
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/models/Employee.java:10]: please undo this import whit a *

File: src/main/java/com/entropyteam/entropay/employees/services/CountryService.java [hunk 3]
```diff
@@ -1,7 +1,8 @@
 package com.entropyteam.entropay.employees.services;
 
-import java.util.UUID;
+import java.util.*;
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/services/CountryService.java:3]: Same, remove imports with *

File: src/main/java/com/entropyteam/entropay/employees/services/CountryService.java [hunk 4]
```diff
@@ -36,4 +39,12 @@ protected CountryDto toDTO(Country entity) {
     protected Country toEntity(CountryDto entity) {
         return new Country(entity);
     }
+
+    @Override
+    public Map<String, Object> getRestrictedFields(AppRole userRole) {
+        Map<String, Object> restrictedFields = new HashMap<>();
+        List<String> countries = countryRepository.findAllByDeletedIsFalseAndNameLikeIgnoreCase(ALL_COUNTRY_NAME).stream().map(Country::getName).toList();
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/services/CountryService.java:46]: minor: i think a better name could be: excluded countries

File: src/main/java/com/entropyteam/entropay/employees/repositories/HolidayRepository.java [hunk 5]
```diff
@@ -19,4 +19,9 @@ public interface HolidayRepository extends BaseRepository<Holiday, UUID> {
     @Query(value = "SELECT DISTINCT extract('Year' FROM date) AS year FROM holiday_calendar WHERE deleted=false "
             + " ORDER BY year ASC", nativeQuery = true)
     List<Integer> getHolidaysYears();
+    @Query(value = "SELECT hc.* FROM holiday_calendar AS hc INNER JOIN country AS c ON hc.country_id = c.id " +
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/repositories/HolidayRepository.java:22]: why not use country id instead of name?

**Review by agustinb4 — CHANGES_REQUESTED** [review]

File: src/main/java/com/entropyteam/entropay/employees/repositories/HolidayRepository.java (hunk 5 above)
Comment (Paez99) [reply src/main/java/com/entropyteam/entropay/employees/repositories/HolidayRepository.java:22]: Changed it to country id

**Review by agustinb4 — APPROVED** [review]
Looks good

**Pull request merged**